core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -122,25 +121,21 @@ object EntityDecoder extends EntityDecoderInstances {
   /** Create a new [[EntityDecoder]]
     *
     * The new [[EntityDecoder]] will attempt to decode messages of type `T`
-    * only if the [[Message]] satisfies the provided [[MediaRange]]s
+    * only if the [[Message]] satisfies the provided [[MediaRange]]
+    * note: Sync bound to note that such an `F[_]` may throw
     */
-  def decodeBy[F[_]: Applicative, T](r1: MediaRange, rs: MediaRange*)(
+  def decodeBy[F[_]: Sync, T](r1: MediaRange, rs: MediaRange*)(
```

Comment:
The `DecodeResult` still lets us `raiseError`, no?

Reply:
I guess but `raiseError` on `EitherT` doesn't result in the same as `F.raiseError`, I think.

Reply:
Should we not tighten this to a sync bound? I'm just not to keen on just `Applicative` possibly denoting something impure (which people can do in the `f`

Reply:
A failure to decode should be a `DecodeResult.failure`, and people can write an `f` that throws regardless of the type bound.  How is that stronger constraint going to be used?